### DISCUSSION TITLE
YARP_FEATURE_XXX

### DISCUSSION THREAD

**Discussion by triccyx** [robotology/yarp#2579] [Ideas]
Speaking of ```yarp::dev::IFrameGrabberControls``` and in particular to set/get/hasFeature
I think it could be a good idea to give more freedom to this method. For new cameras, we will need more FEATUREs
that are not present. So:
1) Give me the possibility to specify my own feature with the current methods.
2) Add to the interface, a new set of methods that could be set/get/hasControl and they do not do any operation on the controlid.

**Comment by drdanz** [comment]
I think it would be a good idea if we could also merge the `IFrameGrabberControls` interface with the `IFrameGrabberControlsDC1394`

**Reply by triccyx** [reply]
Ok agree, but this is out of scope for this discussion
